Bob/dotnet/students

bObXm opened the pull request.

bObXm committed:
test/add/.net

bObXm committed:
UpdateStudent

alexandratodea commented on Test/BusinessLayer/Services/StudentsService.cs:
```diff
@@ -0,0 +1,61 @@
+            }
+            catch(Exception) 
+            {
+                throw;
```
a specific error should be thrown.
https://www.loginradius.com/blog/engineering/exception-handling-in-csharp/

*to be applied everywhere it's needed in this file

alexandratodea commented on Test/DataAccessLayer/Repositories/StudentsRepository.cs:
```diff
@@ -0,0 +1,73 @@
+            List<Student> students = await DataSource();
+            if (!students.Exists(student => student.Id == id))
+            {
+                throw new Exception("Nu exista student!");
```
please use a better error handling mechanism here. also, the error messages should be written in English.

*to be applied everywhere it's needed in this file

alexandratodea commented on Test/DataAccessLayer/Repositories/StudentsRepository.cs:
```diff
@@ -0,0 +1,73 @@
+        public async Task<Student> GetStudentByID(int id)
+        {
+            List<Student> students = await DataSource();
+            if (!students.Exists(student => student.Id == id))
```
what would happen in students list is null?

alexandratodea commented on Test/DataAccessLayer/Repositories/StudentsRepository.cs:
```diff
@@ -0,0 +1,73 @@
+            }
+
+            Student student = students.FirstOrDefault(student => student.Id == id);
+            students.Remove(student);
```
asynchronicity is missing here

bObXm committed:
Fixed error handling mechanism
Changed error messages
Fixed error in case of nullable list
Adding asynchronicity
